examples/examples.rs

```diff
@@ -82,37 +80,26 @@ fn examples() -> Result<()> {
     // This API handles variadics using Heterogeneous Lists.  This is one way to
     // call a function with multiple parameters:
 
-    print.call::<_, ()>(
-        hlist!["hello", "again", "from", "rust"],
-    )?;
+    print.call::<_, ()>(("hello", "again", "from", "rust"))?;
 
     // You can bind rust functions to lua as well
 
-    let check_equal = lua.create_function(|lua, args| {
-        // Functions wrapped in lua receive their arguments packed together as
-        // MultiValue.  The first thing that most wrapped functions will do
-        // is "unpack" this MultiValue into its parts.  Due to lifetime type
-        // signature limitations, this cannot be done automatically from the
-        // function signature, but this will be fixed with ATCs.  Notice the use
-        // of the hlist macros again.
-        let hlist_pat![list1, list2] = lua.unpack::<HList![Vec<String>, Vec<String>]>(args)?;
-
+    let check_equal = lua.create_function(|_, (list1, list2): (Vec<String>, Vec<String>)| {
```

Comment:
> We should also use compiletest to make sure this doesn't regress to allowing bad programs to compile (there's another commented-out test somewhere that could be converted, too). Sadly, I haven't found that crate to be very pleasant to use, but it's all we've got for now.

Yep, actually I had just written the following before I read what you said, so definitely in agreement:

```
// Need to use compiletest-rs or similar to make sure these don't compile.
/*
#[test]
fn should_not_compile() {
    let lua = Lua::new();
    let globals = lua.globals();

    // Should not allow userdata borrow to outlive lifetime of AnyUserData handle
    struct MyUserData;
    impl UserData for MyUserData {};
    let userdata_ref;
    {
        let touter = globals.get::<_, Table>("touter").unwrap();
        touter.set("userdata", lua.create_userdata(MyUserData)).unwrap();
        let userdata = touter.get::<_, AnyUserData>("userdata").unwrap();
        userdata_ref = userdata.borrow::<MyUserData>();
    }

    // Should not allow self borrow of lua, it can change addresses
    globals.set("boom", lua.create_function(|_, _| {
        lua.pack(lua.eval::<i32>("1 + 1", None)?)
    })).unwrap();
}
*/
```